### ISSUE TITLE
nID parameter for Darknet model initialization

### ISSUE THREAD

**Issue by flysofast** [Zhongdao/Towards-Realtime-MOT#110]
Hi, I'm trying to benchmark your jde.1088x608.uncertainty pre-trained model on CVPR19 dataset. I noticed that by default, your JDETracker initialize the Darknet model with nID  = 14455. I have 3 questions:
 - Where does this value come from? is it the total number of object identities in the whole joint dataset that you used for training the model? because seemingly there are not that many identities in any of the datasets you mentioned in DATASET_ZOO.
- Is the value important to the performance of the model, since I could still run the tracking with fairly good result (~60% MOTA) with the same default nID value of 14455?
- If I want to train this model on a different dataset, how should I set this value for the model initialization?

**Comment by Zhongdao** [comment]
Yes, the nID parameter is supposed to be the total number of object identities in the whole joint dataset. For convenience (I am lazy), I use a large number 14455 since it is an estimation of the upper bound of the exact ID numbers. 
It seems better to use the exact ID number, but I think it is not a big problem if the number is larger than the exact nID. This is because the nID parameter is only used to initialize the cross-entropy classifier, I think if the classifier has some redundancy classes the performance will not be affected a lot.

Finally, I suggest you use the exact number of identities.

**Comment by Zhongdao** [comment]
By the way, do you test on the CVPR-19 training set or test set? Would you mind sharing me with your results? I would appreciate it.
If you don't want to post it here in this issue, this is my email: wcd17@mails.tsinghua.edu.cn

Many thanks!

**Comment by mheriyanto** [comment]
> Yes, the nID parameter is supposed to be the total number of object identities in the whole joint dataset. For convenience (I am lazy), I use a large number 14455 since it is an estimation of the upper bound of the exact ID numbers.
> It seems better to use the exact ID number, but I think it is not a big problem if the number is larger than the exact nID. This is because the nID parameter is only used to initialize the cross-entropy classifier, I think if the classifier has some redundancy classes the performance will not be affected a lot.
> 
> Finally, I suggest you use the exact number of identities.

Thanks for your explanation @Zhongdao
